### ISSUE TITLE
bug: re #515: showtabline setting does not persist

### ISSUE THREAD

**Issue by rockfruit** [LazyVim/LazyVim#1291]
### Did you check docs and existing issues?

- [X] I have read all the LazyVim docs
- [X] I have searched the existing issues of LazyVim
- [X] I have searched the existing issues of plugins related to this issue

### Neovim version (nvim -v)

latest

### Operating system/version

linux

### Describe the bug

I disabled bufferline and set `vim.opt.showtabline = 0` in options.lua.  Works great.  Doing `:set showtabline` says "0".

Then: if I open a couple of files (using either :e or neotree), the tabline re-appears, with my buffers listed in little tabs.
At that point doing `:set showtabline` shows me that it's been set to "2".  I can set it to 0 again, and something sets it back to 2 when I open a new file.

I know I can start adding autocmd for this but if that's my attitude then I end up with another frankenvim after a while.  I'd much rather just disable it along with the meddling plugin it rode in on.

I grepped through .local/share/nvim for all `showtabline.*2` and disabled those plugins (lualine and a couple others) to no avail. 

> _I know this may be the wrong place to ask for support, that's why I reference #515 above, where it seems that nobody was berated for this behaviour.  I'm flagging this as a LazyVim bug (when in fact it is expected behaviour based on some included plugin) because ... well it presents as a bug.  If there's anywhere else I can go to discuss the issue please let me know, or just close the issue outright, that's fine too_.

### Steps To Reproduce

1. Install a fresh LazyVim.
2. set `vim.opt.showtabline=0` in config.lua
3. open a couple of files
4. showtabline is now "2".

### Expected Behavior

showtabline remains "0".

### Repro

```Lua
Stock standard latest lazyvim installation.
```

**Comment by dpetka2001** [comment]
For starters you mention in your step 2 you set the option in the file `config.lua`. Was that maybe a typo and you meant `~/.config/nvim/lua/config/options.lua` (I'm mentioning the full path to avoid any possible confusion)? Because that's where the options are supposed to be set.
From that aside, I installed a fresh Lazyvim starter and was not able to reproduce. I didn't install any extra plugins. I just disabled `bufferline` like you mentioned. Then set `vim.opt.showtabline = 0` in the `options.lua` file I mentioned before. I then restarted Neovim and opened 4 tabs with different files in several buffers (2-3 buffers in each tab). The tabline never showed up and checking with `set showtabline?` returned always `0` like I set in `options.lua`.

**Comment by abeldekat** [comment]
@dpetka2001, 
I was also not able to reproduce.

Note that setting vim.opt.showtabline = 0 is not even necessary. The default, being "1", is already suitable for the provided steps to reproduce. No tabs will show up, unless a real tab is opened. 

When tabline is set to "2", neovims tabline appears at the top, showing only the name of the current buffer. I do not see buffers listed in little tabs...

@rockfruit
Did you remove all files in the correponding shared, state and cache folders? For example:
```
rm -rf ~/.local/share/nvim ~/.local/state/nvim ~/.cache/nvim
```

With bufferline disabled, I think the only plugin that may possibly set "showtabline" is lualine.

**Comment by dpetka2001** [comment]
@abeldekat I assumed from the initial post and the nature of this issue, that he wants to work with multiple tabs (thus setting the option to 0). Without setting the option, creating new tabs does indeed show the tabline. That was how I tested based on the assumptions I made and therefore set the setting, so that tabline doesn't show up even when you have multiple tabs open.

**Comment by dpetka2001** [comment]
Also for testing purposes I never use the default nvim folder.
1. `mkdir lazytest`
2. git clone https://github.com/LazyVim/starter ~/.config/lazytest
3. NVIM_APPNAME=lazytest nvim

This way you don't need to mess with your default Neovim configuration. I used to have a docker container with vanilla Lazyvim to do the testing, so that I don't mess with my default Neovim configuration, but `NVIM_APPNAME` since v0.9 was a godsent for testing purposes. Never mess with `~/.config/nvim` and related again :smile:

**Comment by abeldekat** [comment]
@dpetka2001,

> Also for testing purposes I never use the default nvim folder.

I totally agree!

> I assumed from the initial post and the nature of this issue...

I understand the assumption. Thanks! My interpretation is based on my own requirements when disabling bufferline and setting vim.opt.showtabline to "0". I don't want to see the buffers in any tabline, but I do sometimes work with tabs. Those window containers I also don't want to see in the tabline...). 
I noticed that the issue mentions the "edit" command in addition to opening files with neotree. 

As bufferline is disabled, the steps described should also "just work" when the showtabline option is kept at the default.

**Comment by dpetka2001** [comment]
@abeldekat As long as you work with one tab only, then yes. But if you create a second and third tab the tabline shows the tabs at the top without setting that option and leaving it with default value.

**Comment by abeldekat** [comment]
@dpetka2001 
I agree, and I was writing the same in this line:

> The default, being "1", is already suitable for the provided steps to reproduce. No tabs will show up, unless a real tab is opened.

In trying to understand and reproduce the issue, I noticed that in the steps described **only** buffers are used, either via the edit command or via neotree. Maybe this observation could narrow the problem down a bit. 
When @rockfruit reproduces the issue I expect the same problem to arise without setting vim.opts.showstatusline.

**Comment by dpetka2001** [comment]
@abeldekat Yes, I totally agree. I wasn't able to reproduce it no matter what. Maybe @rockfruit will be able to give us a different perspective, when he responds back and tests it again. Maybe I misunderstood some of the steps in between and that's why I'm not able to reproduce? No idea for the time being.

**Comment by abeldekat** [comment]
```
return {
    { "akinsho/bufferline.vim", enabled = false },
}
```

I noticed that lazy will indeed mark the plugin as disabled. Thus, the reference to the plugin is valid. 
However, lazy does not offer to clean the plugin!

The reference should be `"akinsho/bufferline.nvim"`
Note the "n" before "vim".

Thanks for your detailed description!

**Comment by dpetka2001** [comment]
Like @abeldekat noticed, I believe the problem is caused by the incorrect name of bufferline. Tried the exact steps in the previous post and am not able to reproduce.

**Comment by rockfruit** [comment]
Yep, spelling error.
I feel sheepish for sending a bad description initially.

So... it's a different and slightly more subtle bug?
IMHO when there are spelling errors, we should either offer to clean, or should not show "Disabled".

Thanks for your help, and for your patience!
I'm delighted

**Comment by abeldekat** [comment]
You're welcome!
